ztp/siteconfig-generator/siteConfig/siteConfig.go

```diff
@@ -27,16 +27,23 @@ func (sc *SiteConfig) GetSiteConfigFieldValue(path string, clusterId int, nodeId
 	for _, key := range keys[1:] {
 		if v.Kind() == reflect.Slice || v.Kind() == reflect.Array {
 			intf := v.Interface()
-			arrClusters, ok := intf.([]Clusters)
 
+			arrClusters, ok := intf.([]Clusters)
 			if ok {
+				if clusterId < 0 || clusterId >= len(arrClusters) {
+					return nil, fmt.Errorf("Cluster ID out of range: %d", clusterId)
+				}
```

Comment:
This (and the following) replaces a potential 'panic' with proper error handling.